src/js/components/FormField/FormField.js

```diff
@@ -451,7 +459,19 @@ const FormField = forwardRef(
         {...outerProps}
         style={outerStyle}
         onFocus={(event) => {
-          setFocus(shouldKeepFocus());
+          setFocus(containsFocus(formFieldRef.current));
+          const formDescendants =
+            formFieldRef.current.getElementsByTagName('*');
+          let focusableDescendants = 0;
+          for (let i = 0; i < formDescendants.length; i += 1) {
+            const child = formDescendants[i];
+            if (isFocusable(child)) {
+              focusableDescendants += 1;
+            }
+            if (focusableDescendants >= 2 && child === document.activeElement) {
+              setFocus(false);
```

Comment:
I would expect a snapshot change if we had a test case covering this, could be add one here -- maybe using Select + Search or DateInput calendar button as the use case?

Reply:
I added a test in the most recent commit for testing focus with DateInput within a FormField